### PR TITLE
repos: Fix external service config overwrite during long syncs

### PR DESCRIPTION
**Pull request by tsenart**
This commit makes it so we don't overwrite the config field (or any
other fields for that matter) at the end of an external service sync
with stale data that has since been updated (while a long sync was
running).

## Test plan

Integration test.

### CODE REVIEW

File: internal/repos/syncer_test.go [hunk 1]
```diff
@@ -2209,3 +2209,64 @@ func setupSyncErroredTest(ctx context.Context, s *repos.Store, t *testing.T,
...
 }
+
+func testSyncDoesNotOverwriteUpdatedConfig(store *repos.Store) func(*testing.T) {
+	return func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		now := time.Now()
+
+		// create fake source (one user and one org)
```
Comment (tsenart) [inline internal/repos/syncer_test.go:2220]:
```suggestion
```

**Review by ryanslade — APPROVED** [review]
Thanks!

File: internal/repos/syncer.go [hunk 2]
```diff
@@ -616,10 +616,27 @@ func (s *Syncer) SyncExternalService(
...
+	// Re-load the service from the DB and update the sync timestamps. We do this
+	// because a sync can take a long time and we'd potentially over-write changes to
+	// other fields like the config that happened during the sync by re-using the same `svc`
+	// loaded at the beginning of the sync.
+	tx, err := s.Store.ExternalServiceStore.Transact(ctx)
+	if err != nil {
+		return errors.Append(errs, errors.Wrap(err, "failed to start tx"))
+	}
+
+	defer func() { errs = errors.Append(errs, tx.Done(err)) }()
```
Comment (camdencheek) [inline internal/repos/syncer.go:629]: @tsenart This line looks fishy to me. We're appending to `errs` on defer, but `errs` is not the return error value. Basically, I think this is ignoring any errors returned from `tx.Done()`.

Comment (camdencheek) [reply internal/repos/syncer.go:629]: Edit: I see further down, we're doing `return errs`, which is implicitly setting `err = errs`, which I _think_ is fine, but I'm struggling to follow the flow of errors in this chunk

Comment (camdencheek) [reply internal/repos/syncer.go:629]: I think this would surface any errors:
```suggestion
	defer func() { err = errors.Append(err, tx.Done(err)) }()
```

**Pull request merged**